src/mapping/Mapping.cpp

```diff
@@ -106,47 +106,63 @@ void Mapping::map(int inputDataID,
     mapConservative(inputDataID, outputDataID);
   } else if (hasConstraint(CONSISTENT)) {
     mapConsistent(inputDataID, outputDataID);
-  } else if (hasConstraint(SCALEDCONSISTENT)) {
+  } else if (hasConstraint(SCALED_CONSISTENT)) {
     mapConsistent(inputDataID, outputDataID);
-    scaleConsistentMapping(inputDataID, outputDataID);
+    scaleConsistentMapping(inputDataID, outputDataID, SCALED_CONSISTENT);
+  } else if (hasConstraint(SCALED_CONSISTENT_VOLUME)) {
+    mapConsistent(inputDataID, outputDataID);
+    scaleConsistentMapping(inputDataID, outputDataID, SCALED_CONSISTENT_VOLUME);
   } else {
     PRECICE_UNREACHABLE("Unknown mapping constraint.")
   }
 }
 
-void Mapping::scaleConsistentMapping(int inputDataID, int outputDataID) const
+void Mapping::scaleConsistentMapping(int inputDataID, int outputDataID, Mapping::Constraint constraint) const
 {
+  PRECICE_ASSERT(hasConstraint(SCALED_CONSISTENT) || hasConstraint(SCALED_CONSISTENT_VOLUME));
+  bool            volumeMode = hasConstraint(SCALED_CONSISTENT_VOLUME);
+  logging::Logger _log{"mapping::Mapping"};
   // Only serial participant is supported for scale-consistent mapping
   PRECICE_ASSERT((not utils::IntraComm::isPrimary()) and (not utils::IntraComm::isSecondary()));
 
   // If rank is not empty and do not contain connectivity information, raise error
-  if ((input()->edges().empty() and (not input()->vertices().empty())) or
-      (((input()->getDimensions() == 3) and input()->triangles().empty()) and (not input()->vertices().empty()))) {
-    logging::Logger _log{"mapping::Mapping"};
-    PRECICE_ERROR("Connectivity information is missing for the mesh {}. "
-                  "Scaled consistent mapping requires connectivity information.",
-                  input()->getName());
-  }
-  if ((output()->edges().empty() and (not output()->vertices().empty())) or
-      (((output()->getDimensions() == 3) and output()->triangles().empty()) and (not output()->vertices().empty()))) {
-    logging::Logger _log{"mapping::Mapping"};
-    PRECICE_ERROR("Connectivity information is missing for the mesh {}. "
-                  "Scaled consistent mapping requires connectivity information.",
-                  output()->getName());
+  int  spaceDimension    = input()->getDimensions() == 3;
+  bool requiresEdges     = (spaceDimension == 2 and !volumeMode);
+  bool requiresTriangles = (spaceDimension == 2 and volumeMode) or (spaceDimension == 3 and !volumeMode);
+  bool requiresTetra     = (spaceDimension == 3 and volumeMode);
+
+  for (mesh::PtrMesh mesh : {input(), output()}) {
+    if (not mesh->vertices().empty()) {
+      if ((requiresEdges and mesh->edges().empty()) or
+          (requiresTriangles and mesh->triangles().empty()) or (requiresTetra and mesh->tetrahedra().empty())) {
+        PRECICE_ERROR("Connectivity information is missing for the mesh {}. "
```

Comment:
```suggestion
        PRECICE_ERROR("Connectivity information is missing for the mesh \"{}\". "
```
Not completely sure if it is needed.